pinot-common/src/main/java/com/linkedin/pinot/common/utils/request/RequestUtils.java

```diff
@@ -251,4 +258,20 @@ private static String extractDerivedColumn(AggregationInfo aggregationInfo, Star
     return starTreeMetadata.getDerivedHllColumnFromOrigin(aggrColumn);
   }
 
+
+  /**
+   * This method returns the value of {@link #USE_STAR_TREE_KEY} boolean flag specified in the debug options
+   * in broker request. If the flag is not specified in the debug options, it returns true.
+   *
+   * @param brokerRequest Broker Request
+   * @return Value of {@link #USE_STAR_TREE_KEY} debug option, or true if not option not specified.
+   */
+  public static boolean isStarTreeEnabledInBrokerRequest(BrokerRequest brokerRequest) {
+    Map<String, String> debugOptions = brokerRequest.getDebugOptions();
+    if (debugOptions == null) {
+      return true;
+    }
+    String useStarTreeString = debugOptions.get(USE_STAR_TREE_KEY);
+    return (useStarTreeString != null) ? Boolean.valueOf(useStarTreeString) : true;
```

Comment:
As per javadoc of Boolean.valueOf, string other than 'true' defaults to false (I tested that locally as well). Either way, this is just a debug option.